### ISSUE TITLE
New input plugin: logstash-input-jmx-pipe

### ISSUE THREAD

**Issue by mpolajnar** [elastic/logstash#6690]
I (or rather "we") have written a JMX input plugin from scratch. It differs in design from the existing logstash-input-jmx greatly, and I believe it to be superior. I would like you to consider its inclusion in the logstash-plugins repository. It seems to work as is (we already use it to monitor our production systems) and the biggest issue I see is that it is missing most essential unit tests (ie. specs). Only the ones that test configuration validation have been written. I hope to have the tests completed in a few weeks, but the review of the plugin itself could be ongoing in the meantime.

https://github.com/mpolajnar/logstash-input-jmx-pipe

As mentioned in its documentation, the key differences in comparison to logstash-input-jmx are:
* It supports MBean notification subscription. We currently use it to monitor garbage collection events, which is crucial to maintain stability of the JVMs, and plan to use it for many other purposes.
* Our plugin has no outside configuration; all configuration is done within the logstash.conf file.
* It spawns no additional threads it would have to manage. Instead, user is required to configure multiple instances of the plugin to monitor multiple JMX endpoints (or even if she wishes to distribute the load to multiple threads.)
* It allows (and indeed requires) original MBean attribute names to be specified, not the snake_cased_ones.
* It allows multiple MBeans to be queried to produce one logstash event, or, alternatively, querying multiple MBeans using wildcards and producing multiple events at each iteration.
* It allows logging values from the MBean object name key-value property pairs.
* It should be more performant due to the use of getAttributes to obtain the values of all the observed attributes of the MBean at once instead of using the getAttribute in a loop, which is what logstash-input-jmx does.

**Comment by suyograo** [comment]
@mpolajnar this looks great! I've created https://github.com/logstash-plugins/logstash-input-jmx-pipe -- can you create an initial PR there? I'll also make you a maintainer for this repo. 

We can deprecate the existing JMX plugin and eventually delete it in favor of this. For now we can leave both as is and add a notice to the existing JMX repo to notify the users.

**Comment by suyograo** [comment]
I added you as a maintainer. /cc @acchen97

**Comment by mpolajnar** [comment]
I agree with everything you said, but I am unable to create a PR from my repo, because they are not related (my repo is not a fork of the logstash-plugins one). Do you have a trick for doing that in your sleeve? I don't have much experience with github.
